tiledb/sm/query/legacy/reader.cc

```diff
@@ -265,19 +265,69 @@ void Reader::reset() {
 /*         PRIVATE METHODS        */
 /* ****************************** */
 
+Status Reader::load_initial_data() {
+  if (initial_data_loaded_) {
+    return Status::Ok();
+  }
+
+  // Load delete conditions.
+  auto&& [st, delete_conditions] = storage_manager_->load_delete_conditions(
+      array_->array_directory(), *array_->encryption_key());
+  RETURN_CANCEL_OR_ERROR(st);
+  delete_conditions_ = std::move(*delete_conditions);
+
+  // Set timestamps variables
+  user_requested_timestamps_ = buffers_.count(constants::timestamps) != 0 ||
+                               delete_conditions_.size() > 0;
+  const bool partial_consol_fragment_overlap =
+      partial_consolidated_fragment_overlap();
+  use_timestamps_ = partial_consol_fragment_overlap ||
+                    !array_schema_.allows_dups() || user_requested_timestamps_;
+
+  // Add partial overlap condition for timestamps, if required.
+  if (partial_consol_fragment_overlap) {
+    RETURN_NOT_OK(add_partial_overlap_condition());
+  }
+
+  // Legacy reader always uses timestamped conditions. As we process all cell
+  // slabs at once and they could be from fagments consolidated with
+  // timestamps, there is not way to know if we need the regular condition
+  // or the timestamped condition. This reader will have worst performance
+  // for deletes.
+  RETURN_CANCEL_OR_ERROR(generate_timestamped_conditions());
+
+  // Make a list of dim/attr that will be loaded for query condition.
+  if (!condition_.empty()) {
+    for (auto& name : condition_.field_names()) {
```

Comment:
It's not clear to me whether we are checking that field names in a condition are valid; if not, could you please add validation, either here or in one of the coming PRs? (I checked here: https://github.com/TileDB-Inc/TileDB/blob/ce208daf7db9ed10cd74a0d746cdaca009e6420a/tiledb/sm/storage_manager/storage_manager.cc#L1608-L1609)